src/coreclr/vm/exceptionhandling.cpp

```diff
@@ -1840,9 +1845,18 @@ CLRUnwindStatus ExceptionTracker::ProcessOSExceptionNotification(
                     // to the JIT_PInvokeBegin and JIT_PInvokeEnd helpers, which push and pop the ICF on the thread. The
                     // ICF is not linked at the method prolog and unlined at the epilog when running R2R code. Since the
                     // JIT_PInvokeEnd helper will be skipped, we need to unlink the ICF here. If the executing method
-                    // has another pinovoke, it will re-link the ICF again when the JIT_PInvokeBegin helper is called
+                    // has another pinvoke, it will re-link the ICF again when the JIT_PInvokeBegin helper is called.
 
-                    if (ExecutionManager::IsReadyToRunCode(((InlinedCallFrame*)pFrame)->m_pCallerReturnAddress))
+                    TADDR returnAddress = ((InlinedCallFrame*)pFrame)->m_pCallerReturnAddress;
+#ifdef USE_PER_FRAME_PINVOKE_INIT
```

Comment:
Is corinfo.h guaranteed to be included here to pickup this `#define`? 

`corinfo.h` is JIT/EE interface so it does not need to be included globally.

Reply:
Is this one of those things we should set in the CMake files?

Reply:
I've explicitly included `corinfo.h`. Given that this behavior is effectively part of the JIT/EE contract (as one changing behavior will generally require corresponding work in the other to avoid crashes), I think `corinfo.h` is a good place for it to live.